### PR TITLE
apps: cleanup unused code

### PR DESCRIPTION
**Pull request by mfojtik**
@tnozicka PTAL, no functional changes, just minor cleanup.

### CODE REVIEW

File: pkg/apps/apis/apps/types.go [hunk 1]
```diff
@@ -492,9 +492,6 @@ const (
 	// PausedConfigReason is added in a deployment config when it is paused. Lack of progress shouldn't be
 	// estimated once a deployment config is paused.
 	PausedConfigReason DeploymentConditionReason = "DeploymentConfigPaused"
-	// ResumedConfigReason is added in a deployment config when it is resumed. Useful for not failing accidentally
-	// deployment configs that paused amidst a rollout.
-	ResumedConfigReason DeploymentConditionReason = "DeploymentConfigResumed"
```
Comment (mfojtik) [inline pkg/apps/apis/apps/types.go]: @kargakis @tnozicka i don't see this used anywhere, why we keeping this API?

**Comment by mfojtik** [comment]
flake: https://github.com/openshift/origin/issues/19001

File: pkg/apps/registry/instantiate/rest.go [hunk 2]
```diff
@@ -50,14 +50,14 @@ func (s *REST) New() runtime.Object {
 }
 
 // Create instantiates a deployment config
-func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, _ bool) (runtime.Object, error) {
+func (s *REST) Create(ctx apirequest.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, _ bool) (runtime.Object, error) {
```
Comment (tnozicka) [inline pkg/apps/registry/instantiate/rest.go]: I'd rather change the other one to `r`, no idea what `s` stands for here; it's usually the first letter

File: pkg/apps/strategy/rolling/rolling.go [hunk 3]
```diff
@@ -163,12 +162,12 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
...
 	if params.Pre != nil {
 		if err := s.hookExecutor.Execute(params.Pre, to, appsapi.PreHookPodSuffix, "pre"); err != nil {
-			return fmt.Errorf("pre hook failed: %s", err)
+			return fmt.Errorf("pre-hook failed: %s", err)
 		}
 	}
 
 	if s.until == "pre" {
-		return strat.NewConditionReachedErr("pre hook succeeded")
+		return strat.NewConditionReachedErr("pre-hook succeeded")
```
Comment (tnozicka) [inline pkg/apps/strategy/rolling/rolling.go]: Isn't this changing our "api" to the user? I know people are grepping these - do we have a good reason to mess with it?

Comment (tnozicka) [reply pkg/apps/strategy/rolling/rolling.go]: there may be some temporarily disabled check for it in tests due to logging issues

File: pkg/apps/registry/instantiate/rest.go (hunk 2 above)
Comment (mfojtik) [reply pkg/apps/registry/instantiate/rest.go]: s as strategy

File: pkg/apps/strategy/rolling/rolling.go (hunk 3 above)
Comment (mfojtik) [reply pkg/apps/strategy/rolling/rolling.go]: error messages is not an api

Comment (tnozicka) [reply pkg/apps/strategy/rolling/rolling.go]: that's why the quotes ;)

but this is what oc deploy, rollout and logs will return to the user and they are grepping it

We are allowed to break it, we should just do it intentionally and with some gain.

File: pkg/apps/apis/apps/helpers.go [hunk 4]
```diff
@@ -89,28 +89,11 @@ func EachTemplateImage(pod *kapi.PodSpec, triggerFn TriggeredByFunc, fn func(Tem
...
-	}
-	return nil
-}
-
 // TriggeredByFunc returns a TemplateImage or error from the provided container
 type TriggeredByFunc func(container *kapi.Container) (TemplateImage, bool)
 
 // IgnoreTriggers ignores the triggers
-func IgnoreTriggers(container *kapi.Container) (TemplateImage, bool) {
+func IgnoreTriggers(_ *kapi.Container) (TemplateImage, bool) {
```
[the remark this answers is not in the document]
Comment (mfojtik) [reply pkg/apps/apis/apps/helpers.go]: govet complains in goland when the parameter is not used... i thin I agree that we can keep it for the sake of docs

**Comment by mfojtik** [comment]
@tnozicka updated, ptal again

**Pull request merged**